### PR TITLE
strip trailing blank lines for some templated audit rules

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- strip trailing blank lines for some rules using audit_file_contents template.

#### Rationale:

- some rule snippets provided as part of files using audit_file_contents template have trailing lines, some do not. This PR tries to align the content with files provided by Audit package.

### CODE REVIEW

**Comment by yuumasato** [comment]
@vojtapolasek Is there any difference in the built DS? Could you clarify which rule is expected to have the trailing spaces trimmed?
I could not identify any content difference with and without this PR.

**Comment by yuumasato** [comment]
I see the difference now, and have a few questions.
I see that these rules had `|+` for 2 years, and now they are broken. Do you know what caused this? 

I also see other rulres with `|+` instead of `|-`, I guess they should be changed too?
```
linux_os/guide/system/auditing/policy_rules/audit_basic_configuration/rule.yml-        contents: |+
linux_os/guide/system/auditing/policy_rules/audit_immutable_login_uids/rule.yml-        contents: |+
linux_os/guide/system/auditing/policy_rules/audit_ospp_general/rule.yml-        contents: |+

```

**Comment by vojtapolasek** [comment]
Hello,
the change manifested because there were added more lines (fix text) below the definition of template. The "+" does not strip trailing new lines, so it "ate" the new line used as separator in the file.
I will try to fix also other rules you mention. Note that they DO require a trailing blank line to align with the file provided by Audit.

**Comment by vojtapolasek** [comment]
Hello, I believe I fixed it. I also updated commit messages for more clarity.

File: linux_os/guide/system/auditing/policy_rules/audit_basic_configuration/rule.yml [hunk 1]
```diff
@@ -73,4 +73,5 @@ template:
     vars:
         filepath: /etc/audit/rules.d/10-base-config.rules
         contents: |+
-            {{{ file_contents_audit_base_config|indent(12) }}}
+            {{{ file_contents_audit_base_config|indent(12) }}}#do not remove this comment
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/policy_rules/audit_basic_configuration/rule.yml:76]: Is it expected that this line contains a comment?
Does it need to be trailing the Jinja code?

**Comment by vojtapolasek** [comment]
The comment is not shown within the templated check / remediation it self. It just stops Jinja from including more trailing blank lines. It is not nice, but I am not able to find a better solution.
Another workaround would be to add there a comment below stating that the line MUST NOT be followed by more blank lines or something like that... What do you think?

**Comment by vojtapolasek** [comment]
I understand. I fixed it.

File: linux_os/guide/system/auditing/policy_rules/audit_basic_configuration/rule.yml [hunk 2]
```diff
@@ -74,3 +73,4 @@ template:
         filepath: /etc/audit/rules.d/10-base-config.rules
         contents: |+
             {{{ file_contents_audit_base_config|indent(12) }}}
+#do not remove this comment, it stops Jinja from including more blank lines to the variable
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/policy_rules/audit_basic_configuration/rule.yml:76]: Similarly, it seems that the remediation adds one blank line to the file, but the OVAL doesn't want it.

**Comment by yuumasato** [comment]
Un-assigning my self as I'll be out for a few days. Somebody please take over the review.

**Comment by yuumasato** [comment]
I had some changes on my local repo that were affecting the rules.
Testing again they are working fine.

**Review by yuumasato — APPROVED** [review]

**Pull request merged**